api/v1/db/oracledb/onbase-dao.js

```diff
@@ -0,0 +1,107 @@
+const appRoot = require('app-root-path');
+const createError = require('http-errors');
+const _ = require('lodash');
+const { BIND_OUT, NUMBER, STRING } = require('oracledb');
+
+const { serializeOnBase } = require('../../serializers/onbase-serializer');
+
+const conn = appRoot.require('api/v1/db/oracledb/connection');
+const { contrib } = appRoot.require('api/v1/db/oracledb/contrib/contrib');
+
+
+/**
+ * @summary A Helper recursive function to read buffer
+ * @function
+ * @returns {Promise<String[]>} Promise object represents a list of records
+ */
+const getLine = async (connection, lines) => {
+  try {
+    const { outBinds } = await connection.execute(
+      contrib.getLine(),
+      {
+        line: { dir: BIND_OUT, type: STRING, maxSize: 32767 },
+        status: { dir: BIND_OUT, type: NUMBER },
+      },
+    );
+    if (outBinds.line) {
+      lines.push(outBinds.line);
+    }
+    // The status code will be equal to 1 if there is no more output
+    if (outBinds.status !== 1) {
+      ({ connection, lines } = await getLine(connection, lines));
```

Comment:
Yes, it is. The parentheses here are mainly for fixing the issue of destructuring assignment since we're reassigning values to the arguments of this function.

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Syntax